README.md

```diff
@@ -10,6 +10,60 @@ If there is anything you are looking for that is not covered here you should ref
 
 ----
 
+### File structure ###
+
+You should not have multiple types in the same file, each type should have its own file.
+
+The following list should be the standard struct/class organization of all your Swift files, in this specific order:
+
+Before the class declaration:
+
+* Enums (as private)
+
+Inside the class:
+
+* Public properties
+* Private properties
+* Public static variables
+* Private static variables
+* Init/Deinit
+* Override functions
+	* View lifecycle if in a view controller
+	* Other override functions 
+* Public functions
+* Private functions
+
+After the class declaration:
+
+* Extensions
+
+When implementing a protocol you should create an extension of your class that lives in the same file to separate the core logic of your class and your protocol implementation.
+
+#### Enum ####
+
+All enums should live in their own code file, except in cases where the enum is declared as private. In cases where the enum is declared private, declare the enum at the top of the code file, above the type declaration.
+
+#### Usage of MARK ####
+
+To help organizing your files you may want to use pragma marks to clearly separate your functions, properties, extensions. Only use them for the groups defined above. Also use a MARK per extensions.
+
+```swift
+// MARK: Public Functions
+
+public func doSomething() {
+}
+
+// MARK: Internal Functions
+
+func doSomething() {
```

Comment:
Yes, for all top-level declarations. That rule is already merged in. If it's a top-level declaration (in global scope, not within the scope of a type), it must have an access modifier. This is also in Github's style guide and is a best practice, imo.